docs/guides/directing-traffic-to-your-app.md

```diff
@@ -0,0 +1,375 @@
+
+![coffee app](/docs/images/direct-all-traffic-app.png)
+
+With this architecture, the coffee application is not accessible outside the cluster. We want to expose this application
+on the hostname `cafe.example.com` so that users outside the cluster can access it.
+
+To do this, we will install NGINX Kubernetes Gateway and create two Gateway API resources: a Gateway and an HTTPRoute.
+With these resources, we will configure a simple routing rule to match all HTTP traffic with the
+hostname `cafe.example.com` and direct it to the coffee Service.
+
+## Setup
+
+Create the coffee application in Kubernetes by copying and pasting the following into your terminal:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: coffee
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: coffee
+  template:
+    metadata:
+      labels:
+        app: coffee
+    spec:
+      containers:
+      - name: coffee
+        image: nginxdemos/nginx-hello:plain-text
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: coffee
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+    protocol: TCP
+    name: http
+  selector:
+    app: coffee
+EOF
+```
+
+This will create the coffee Service and a Deployment with two Pods. Run the following command to verify the resources
+were created:
+
+```shell
+kubectl get pods,svc
+```
+
+Your output should include 2 coffee Pods and the coffee Service:
+
+```text
+NAME                          READY   STATUS      RESTARTS   AGE
+pod/coffee-7dd75bc79b-cqvb7   1/1     Running     0          77s
+pod/coffee-7dd75bc79b-dett3   1/1     Running     0          77s
+
+
+NAME                 TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
+service/coffee       ClusterIP   10.96.75.77     <none>        80/TCP    77s
+```
+
+## Application Architecture with NGINX Kubernetes Gateway
+
+To direct traffic to the coffee application, we will create a Gateway and HTTPRoute. The following diagram shows the
+configuration we'll be creating in the next step:
+
+![Configuration](/docs/images/direct-all-traffic-config.png)
+
+We need a Gateway to create an access point for HTTP traffic coming into the cluster. The `cafe` Gateway we are going to
+create will open an access point to the cluster on port 80 for HTTP traffic.
+
+To direct HTTP traffic from the Gateway to the coffee Service, we need to create an HTTPRoute named `coffee` and attach
+to the Gateway. This HTTPRoute will have a single routing rule that directs all traffic to the
+hostname `cafe.example.com` from the Gateway to the coffee Service.
+
+Once NGINX Kubernetes Gateway parses the `cafe` Gateway and `coffee` HTTPRoute, it will configure its dataplane, NGINX,
+to route all HTTP requests to `cafe.example.com` to the Pods that the `coffee` Service targets:
+
+![Traffic Flow](/docs/images/direct-all-traffic-flow.png)
```

Comment:
I suggest replacing 'http://cafe.example.com' with 'XXX.YYY.ZZZ.III` because the NKG exposed via that API.

We can annotation the arrow from clients to public endpoint with `http://cafe.example.com` to add more context.

Reply:
Changed to Public IP Address and kept the hostname in the box based on this discussion: https://github.com/nginxinc/nginx-kubernetes-gateway/pull/993#discussion_r1300677827

Reply:
Can we drop "http://" prefix ? For example, we can have "Public IP Address for cafe.example.com" - since http is not involved in converting hostname to IP. I think it is a little confusing with http:// prefix